docs/storage.md

```diff
@@ -332,8 +332,28 @@ config:
   container: ""
   endpoint: ""
   max_retries: 0
+  msi_resource: ""
+  pipeline_config:
+    max_tries: 0
+    try_timeout: 0s
+    retry_delay: 0s
+    max_retry_delay: 0s
+  reader_config:
+    max_retry_requests: 0
+  http_config:
+    idle_conn_timeout: 0s
+    response_header_timeout: 0s
+    insecure_skip_verify: false
```

Comment:
Could [tls_config](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#tls_config) be added instead of just `insecure_skip_verify`?
I know blob storage itself does not support it but we'd like to use a proxy (by changing the `endpoint` in the configuration) with mTLS for blob storage to avoid having to provide the storage account key to every cluster we monitor.
I also think it would be better to make the config as similar to Prometheus as possible for consistency and code resuse (if possible)

Reply:
yes and no. This is in line with all other storage options in Thanos: https://thanos.io/tip/thanos/storage.md/
There is some work being done to make these configs more logical, but for now I try to keep it aligned with how everything in Thanos works (:

Reply:
thanks for the answer, I guess the solution to this problem is to implement a custom proxy and run it as a sidecar, using https with a dummy certificate and `insecure_skip_verify`

Reply:
> thanks for the answer, I guess the solution to this problem is to implement a custom proxy and run it as a sidecar, using https with a dummy certificate and `insecure_skip_verify`

Yea I understand the case, it's just that I don't feel making one specific change here. However, it would make sense to implement a generic tls_config for non-component related configs, but for 'object storage connections'. As this would/should be implemented not only for Azure Blob, but also for all the other options Thanos supports. 
Perhaps you could create a new issue for this?